server.js

```diff
@@ -0,0 +1,170 @@
+  User.findAll({ where: { id: req.params.id } })
+    .then(users => res.send(users))
+    .catch(next);
+});
+
+app.put('/api/users/:id', (req, res, next) => {
+  User.findByPk(req.params.id)
+    .then(_user =>
+      _user.update({
+        username: req.body.username,
+        email: req.body.email,
+        password: req.body.password,
+        firstName: req.body.firstName,
+        lastName: req.body.lastName,
+        streetAddress: req.body.streetAddress,
+        city: req.body.city,
+        state: req.body.state,
+        zipcode: req.body.zipcode,
+        billStreetAddress: req.body.billStreetAddress,
+        billCity: req.body.billCity,
+        billState: req.body.billState,
+        billZipcode: req.body.billZipcode
+      })
+    )
+    .then(() => res.sendStatus(201))
+    .catch(next);
+});
+
+app.get('/api/products', (req, res, next) => {
+  Product.findAll()
+    .then(products => res.send(products))
+    .catch(next);
+});
+
+app.get('/api/orders', (req, res, next) => {
+  Order.findAll()
+    .then(orders => res.send(orders))
+    .catch(next);
+});
+
+app.get('/api/orderProducts', (req, res, next) => {
+  OrderProducts.findAll()
+    .then(orders => res.send(orders))
+    .catch(next);
+});
+
+app.post('/api/orderProducts', async (req, res, next) => {
+  Order.findOne({
+    where: {
+      status: 'cart',
+      userId: req.body.userId
+    }
+  })
+    .then(async order => {
+      const item = await OrderProducts.create({ ...req.body, orderId: order.id })
+      // console.log('server item', item)
+      res.send(item);
+    })
+    .catch(err => next(err));
+});
+
+app.delete('/api/orderProducts/:id', async (req, res, next) => {
+  await OrderProducts.destroy({ where: { id: req.params.id } });
+  res.sendStatus(204);
+});
+
+app.put('/api/orderProducts/:id', async (req, res, next) => {
+  const item = await OrderProducts.update(
+    { quantity: req.body.quantity },
+    { where: { id: req.body.id } }
+  );
+  console.log(item);
+  res.send(item);
+});
+
+// Login
+
+app.post('/api/sessions', (req, res, next) => {
+  User.findOne({
+    where: {
+      email: req.body.email,
+      password: req.body.password
+    }
+  })
+    .then(user => {
+      if (!user) {
+        throw { status: 401 };
+      }
+      req.session.user = user;
+      return res.send(user);
+    })
+    .catch(err => next(err));
+});
+
+app.get('/api/sessions', (req, res, next) => {
+  const user = req.session.user;
+  if (user) {
+    return res.send(user);
+  }
+  next({ status: 401 });
+});
+
+app.delete('/api/sessions', (req, res, next) => {
+  req.session.destroy();
+  req.session = null;
+  res.sendStatus(204);
+});
+
+// Page Not Fount Route
+app.get('*', (req, res) => {
+  res.send(`
+    <h1>404 Page Not Found</h1>
+    <p>Sorry, that page doesn't exist, Doc. :(</p>
+  `);
+});
+
+db.syncAndSeed().then(() => {
+  app.listen(port, () => console.log(`listening on port ${port}`));
+});
+
+module.exports = app;
```

Comment:
Routes are pretty clean otherwise, need to do some sanity checks on incoming data to ensure they fit the schema required to do the proper database work, and solve some security concerns around exposing user data, but as a whole it is a clean server file. You may want to consider breaking this up by type of api route into separate files.